e2e/real/sessions.spec.ts

```diff
@@ -0,0 +1,180 @@
+  // open up popup
+  await Promise.all([
+    page.waitForEvent('popup'),
+    page.getByRole('button', { name: 'Create a plot' }).click(),
+  ]);
+
+  await page.getByRole('button', { name: 'Save as' }).click();
+
+  await page.getByRole('textbox', { name: 'Name' }).fill('e2e testing session');
+  await page
+    .getByRole('textbox', { name: 'Summary' })
+    .fill('test session summary');
+
+  // listen for sessions response to get the ID of the session we create
+  // so we can ensure it's tidied up later
+  const responsePromise = page.waitForResponse('**/sessions?*');
+
+  await page.getByRole('button', { name: 'Save' }).click();
+
+  const response = await responsePromise;
+  sessionId = await response.json();
+
+  await expect(
+    page.getByRole('dialog', { name: 'Save Session' })
+  ).not.toBeVisible();
+
+  // reload page
+  await page.reload();
+
+  const popups: (typeof page)[] = [];
+  page.on('popup', async (popup) => {
+    await popup.waitForLoadState();
+    popups.push(popup);
+  });
+
+  // load session
+  await page
+    .getByRole('button', { name: 'e2e testing session', exact: true })
+    .click();
+
+  await expect(page.getByRole('progressbar')).toBeVisible();
+  await expect(page.getByRole('progressbar')).not.toBeVisible();
+
+  await expect(popups).toHaveLength(3);
+
+  // edit session metadata
+
+  await page
+    .getByRole('button', { name: 'edit e2e testing session session' })
+    .click();
+
+  await page
+    .getByRole('textbox', { name: 'Name' })
+    .fill('renamed e2e testing session');
+
+  await page.getByRole('button', { name: 'Save' }).click();
+
+  await expect(
+    page.getByRole('button', {
+      name: 'renamed e2e testing session',
+      exact: true,
+    })
+  ).toBeVisible();
+  await expect(
+    page.getByRole('button', { name: 'e2e testing session', exact: true })
+  ).not.toBeVisible();
+
+  // edit session (change rows per page)
+
+  await page
+    .getByRole('combobox', {
+      name: 'Rows per page',
+    })
+    .click();
+
+  await page
+    .getByRole('option', {
+      name: '50',
+    })
+    .click();
+
+  await page.getByRole('button', { name: 'Save', exact: true }).click();
+
+  await page.reload();
+
+  await page
+    .getByRole('button', { name: 'renamed e2e testing session', exact: true })
+    .click();
+
+  await expect(
+    page.getByRole('combobox', {
+      name: 'Rows per page',
+    })
+  ).toHaveText('50');
+
+  // delete session
+  await page
+    .getByRole('button', { name: 'delete renamed e2e testing session session' })
+    .click();
+
+  await page.getByRole('button', { name: 'Continue' }).click();
+
+  await expect(
+    page.getByRole('dialog', { name: 'Delete Session' })
+  ).not.toBeVisible();
+
+  await expect(
+    page.getByRole('button', {
+      name: 'renamed e2e testing session',
+      exact: true,
+    })
+  ).not.toBeVisible();
+
+  // if we get to the end of the test, no need to our after each to delete the session
```

Comment:
This comment doesn't make sense.

Reply:
Good catch, took me a second to think of what I was meaning to say with that comment: `if we get to the end of the test, no need to run our after each code to delete the session (since we just deleted it as part of the test), so reset session id`. I'll update it in a sec